### ISSUE TITLE
s07 parquet memory: column-prune + categorical (10.4GB → 5.1GB on MK)

### ISSUE THREAD

**Issue by hazeydata** [hazeydata/theme-park-crowd-report#479]
## Context

Surfaced during S20 / TPCR #470 proof-batch OOM diagnostic. Wilma's pre-promotion crash on the day_of_week proof batch was rooted here.

## The bug

\`pipeline/steps/s07_training.py\` calls \`pd.read_parquet(park_file)\` without column pruning. MK.parquet has **39.2M rows × 15 columns**, including 6 string columns (\`park_date\`, \`entity_code\`, \`date_group_id\`, \`season\`, \`season_year\`, \`observed_at\`). pandas materializes string columns as object dtype at ~80 bytes/cell → **~10.4 GB RSS just to hold MK in memory before any training starts**. DL.parquet (608 MB on disk vs MK's 539 MB) extrapolates to ~11.7 GB.

## Proposed fix (verified in proof-batch harness)

Two-line change in s07's outer loop:

```python
NEEDED_COLS = [
    "entity_code", "park_date", "actual_time",
    "mins_since_6am", "mins_since_open",
    "date_group_id_encoded", "season_encoded", "season_year_encoded",
    "is_synthetic",
]
park_df = pd.read_parquet(park_file, columns=NEEDED_COLS)  # was: pd.read_parquet(park_file)
park_df["entity_code"] = park_df["entity_code"].astype("category")  # new
```

## Verified impact (MK)

- Before: 10,400 MB park_df peak (single-park concat)
- After: **5,112 MB park_df peak** (col-prune)
- After + categorical: pd_memory deep ~3,444 MB (categorical saves ~1.9 GB on entity_code alone)
- Per-entity train footprint: avg +93 MB first call (xgb warmup), +0–16 MB steady after
- Extrapolation to 420 entities at DL-scale: **~14 GB peak total** (vs estimated 20+ GB without fix)

Full diagnostic in #470 proof-batch comments.

## Scope guard

- **Mechanical change.** Same dtypes, same hyperparameters, same eval split, same model output.
- Should produce bit-identical models to the current pipeline (XGBoost is deterministic on the same input matrix).
- Rule 17 verification: train one entity (e.g. MK05) before/after, check \`model_baseline.json\` MD5 matches.

## Out of scope

- s08_forecast.py read paths (separate ticket if needed).
- Memory profiling of TDL/TDS/EU (smaller parquets, less urgency).

🪖 Filed per #470 close-out instruction.
